rust/bridge/jni/bin/gen_java_decl.py

```diff
@@ -13,31 +13,45 @@
 
 cbindgen = subprocess.Popen(['cbindgen'], cwd=os.path.join(our_abs_dir, '..'), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
 
-(stdout, _) = cbindgen.communicate()
+(stdout, stderr) = cbindgen.communicate()
 
 stdout = str(stdout.decode('utf8'))
+stderr = str(stderr.decode('utf8'))
+
+ignore_this_warning = re.compile("WARN: Can't find .*\. This usually means that this type was incompatible or not found\.")
+for l in stderr.split('\n'):
+    if l == "":
+        continue
+
+    if ignore_this_warning.match(l):
+        continue
+
+    print(l)
 
 java_decl = re.compile('([a-zA-Z]+) Java_org_signal_client_internal_Native_([A-Z][a-zA-Z]+)_1([A-Za-z0-9]+)\(JNIEnv .?env, JClass class_(, .*)?\);')
 
 def translate_to_java(typ):
-    if typ == "void":
-        return "void"
-    if typ == "jstring" or typ == "JString":
-        return "String"
-    if typ == "jbyteArray":
-        return "byte[]"
-    if typ == "ObjectHandle":
-        return "long"
-    if typ == "jint":
-        return "int"
-    if typ == "jlong":
-        return "long"
-    if typ == "jboolean":
-        return "boolean"
-    if typ == "jobject":
-        return "Object"
-
-    raise Exception("Don't know what to do with a", typ)
+    # jobject is not given here; instead use a type
+    type_map = {
+        "void": "void",
+        "jstring": "String",
+        "JString": "String",
+        "jbyteArray": "byte[]",
+        "ObjectHandle": "long",
+        "jint": "int",
+        "jlong": "long",
+        "jboolean": "boolean",
+        "JavaSessionStore": "SessionStore",
+        "JavaIdentityKeyStore": "IdentityKeyStore",
+        "JavaPreKeyStore": "PreKeyStore",
+        "JavaSignedPreKeyStore": "SignedPreKeyStore",
+        "JavaSenderKeyStore": "SenderKeyStore",
+        "JavaCiphertextMessage": "CiphertextMessage",
```

Comment:
Can we pattern-match these? "If the type starts with 'Java', drop that and assume it's valid."

Reply:
Makes sense. Each type still has to be named in Native.java.in so it's not much of a savings but every bit helps.